hyperbook/book/spiele/mississippi-queen/xml.md

```diff
@@ -0,0 +1,262 @@
+
+- Die ``<state>``-Ebene gibt Auskunft über den aktuellen Spielzustand, einschließlich des Spielzugs (``turn``) und des Teams (``currentTeam``), das gerade am Zug ist.
+
+- Die ``<board>``-Ebene enthält Informationen über das Spielfeld, darunter die erwartete nächste Richtung (``nextDirection``), in die ein Segment erweitert werden kann.
+
+- Die ``<segments>``-Ebene enthält eine Liste von Segmenten, die jeweils eine bestimmte Richtung und ein Zentrum haben. Jedes Segment kann verschiedene Feldtypen enthalten, wie Wasser, Inseln, Passagiere usw., die in Arrays organisiert sind.
+:::alert{info}
+Das Passagier-Feld nimmt eine besondere Stellung gegenüber den anderen Feldern ein, da dieses eine Richtung und eine Anzahl und Passagieren hat. Dies ist nötig, um dem Spieler mitzuteilen, von wo aus dieser die Passagiere aufnehmen kann.
+:::
+
+- Die ``<ship>``-Ebene enthält Informationen über Schiffe im Spiel. Jedes Schiff wird durch Teamzugehörigkeit (``team``), Punktzahl (``points``), Blickrichtung (``direction``), Geschwindigkeit (``speed``), Kohlebestand (``coal``), Anzahl der Passagiere (``passengers``) und verbleibende Runden mit freier Bewegung (freeTurns) charakterisiert. Die Position des Schiffs wird durch [Cube-Koordinaten](https://www.redblobgames.com/grids/hexagons/#coordinates-cube) ($q$, $r$ und $s$) angegeben.
+
+Dieses XML-Protokoll bietet eine umfassende Struktur zur Modellierung eines hexagonalen Spielfelds mit Segmenten, die in einem übergeordneten Koordinatensystem platziert sind. Die Verwendung von [Cube-Koordinaten](https://www.redblobgames.com/grids/hexagons/#coordinates-cube) ermöglicht eine präzise Positionierung und Navigation im Spielfeld.
+
+#### Spielablauf
+
+Der erste Spieler erhält dann eine Zugaufforderung:
+
+```xml
+<room roomId="ROOM_ID">
+  <data class="moveRequest" />
+</room>
+```
+
+Worauf dieser innerhalb der gesetzten Zeitbeschränkung mit einem der folgenden Zug-Typen antwortet:
+
+##### Spielzug
+Ein Zug kann beispielhaft wie folgt aussehen:
+```xml
+<room roomId="ROOM_ID">
+  <move>
+      <actions>
+          <acceleration acc="1" />
+          <advance distance="2" />
+          <push direction="RIGHT" />
+          <turn direction="DOWN_RIGHT" />
+      </actions>
+  </move>
+</room>
+```
+
+Ein Zug besteht immer aus einer Liste aus Aktionen. Die Reihenfolge dieser Liste bestimmt auch, in welcher Reihenfolge die Aktionen ausgeführt werden. Das ist besonders wichtig zu beachten, da einige Aktionen nur erlaubt sind, wenn andere bereits ausgeführt wurden, oder Aktionen wie, die Beschleunigungsaktion, ist stets immer als erste Aktion erlaubt.
+
+##### Weiteres
+
+Nach Erhalt des Zuges sendet der Server den neuen Spielstatus an alle Spieler 
+und dem nächsten Spieler eine Zugaufforderung.
+So geht es hin und her bis zum [Spielergebnis](#spielergebnis).
+
+#### Ende der Kommunikation
+
+Die letzte Nachricht des Servers endet mit:
+
+```xml
+</protocol>
+```
+
+Danach wird die Verbindung geschlossen.
```

Comment:
Warum ist das in der spielspezifischen Sektion?

Reply:
Da dies in den letzten Jahren auch so gemacht wurde.

Reply:
Ja, und ich fand es schon damals nicht gut :sweat_smile: